sdk/communication/azure-communication-callingserver/src/test/java/com/azure/communication/callingserver/DownloadContentAsyncLiveTests.java

```diff
@@ -186,6 +186,28 @@ public void downloadContent404Async(HttpClient httpClient) {
             () -> FluxUtil.collectBytesInByteBufferStream(response.getValue()).block());
     }
 
+    @ParameterizedTest
+    @MethodSource("com.azure.core.test.TestBase#getHttpClients")
+    @DisabledIfEnvironmentVariable(
+        named = "SKIP_LIVE_TEST",
+        matches = "(?i)(true)",
+        disabledReason = "Requires human intervention")
+    public void downloadMetadataWithRedirectAsync(HttpClient httpClient) {
+        CallingServerClientBuilder builder = getConversationClientUsingConnectionString(httpClient);
+        CallingServerAsyncClient conversationAsyncClient = setupAsyncClient(builder, "downloadMetadataAsync");
+
+        try {
+            Flux<ByteBuffer> content = conversationAsyncClient.downloadStream(METADATA_URL);
```

Comment:
This could be done using Reactor's test framework `StepVerifier`. https://projectreactor.io/docs/test/release/api/reactor/test/StepVerifier.html